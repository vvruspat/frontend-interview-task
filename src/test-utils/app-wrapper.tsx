import React, { FC } from "react";
import { Provider } from "react-redux";
import { GlobalStyles } from "../components/global-styles";
import { AuthProvider } from "../auth";
import { store } from "../store/store";

export const AppWrapper: FC = ({ children }) => {
  return (
    <Provider store={store}>
      <GlobalStyles />
      <AuthProvider>{children}</AuthProvider>
    </Provider>
  );
};
