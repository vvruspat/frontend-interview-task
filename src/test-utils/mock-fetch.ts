import { error401Data } from "./test-data/error";
import { feedbacksData } from "./test-data/feedbacks";
import { jwtData } from "./test-data/jwt";
import { themesData } from "./test-data/themes";

export function mockFetch() {
  let status: number = 200;

  return {
    setStatus: (newStatus: number) => {
      status = newStatus;
    },

    fetch: jest.fn().mockImplementation((url: string) => {
      if (status !== 200) {
        return Promise.reject(error401Data);
      }

      if (url.indexOf("/api/reviews") !== -1) {
        return Promise.resolve({
          ok: true,
          status: status,
          json: () => Promise.resolve({ data: feedbacksData }),
        });
      }

      if (url.indexOf("/api/themes") !== -1) {
        return Promise.resolve({
          ok: true,
          status: status,
          json: () => Promise.resolve({ data: themesData }),
        });
      }

      if (url.indexOf("/login") !== -1) {
        return Promise.resolve({
          ok: true,
          status: status,
          json: () => Promise.resolve(jwtData),
        });
      }

      return Promise.resolve({
        ok: true,
        status: status,
        json: () => {},
      });
    }),
  };
}
