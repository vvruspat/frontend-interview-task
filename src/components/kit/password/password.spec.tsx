import React from "react";
import { fireEvent, render } from "@testing-library/react";
import { AppWrapper } from "../../../test-utils/app-wrapper";
import { Password } from ".";

describe("Password", () => {
  test("renders password", async () => {
    const onChange = jest.fn();
    const component = render(
      <Password onChange={onChange} data-testid="password-input" />,
      { wrapper: AppWrapper },
    );
    const password = component.getByTestId("password-input");
    const hidePasswordButton = component.getByTestId("hide-password-button");

    fireEvent.change(password, { target: { value: "test" } });

    expect(password).toBeInTheDocument();
    expect((password as HTMLInputElement).value).toBe("test");
    expect(password).toMatchSnapshot();
    expect(onChange).toBeCalled();
    expect((password as HTMLInputElement).type).toBe("password");

    fireEvent(
      hidePasswordButton,
      new MouseEvent("click", {
        bubbles: true,
        cancelable: true,
      }),
    );

    expect((password as HTMLInputElement).type).toBe("text");
  });
});
