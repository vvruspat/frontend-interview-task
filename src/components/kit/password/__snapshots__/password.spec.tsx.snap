// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Password renders password 1`] = `
<input
  class="sc-bdfBwQ sc-dlfnbm gcJGpO gOrIRV"
  data-testid="password-input"
  type="password"
/>
`;
